### PR TITLE
Fix huber loss backward for mixed inputs

### LINKED ISSUE #76646: `torch.nn.HuberLoss` backwards unexpectedly fail
### 🐛 Describe the bug

`torch.nn.HuberLoss` backwards unexpectedly fail with error message `Found dtype Float but expected Double`. 
For its documentation, "When delta is set to 1, this loss is equivalent to [SmoothL1Loss](https://pytorch.org/docs/stable/generated/torch.nn.SmoothL1Loss.html#torch.nn.SmoothL1Loss)". As a result, it should succeed to backward like `SmoothL1Loss`

```python
import torch
t1_tensor = torch.rand([67, 1], dtype=torch.float64)
t2_tensor = torch.rand([67, 1], dtype=torch.float32)

t1 = t1_tensor.clone().detach().requires_grad_().to('cpu')
t2 = t2_tensor.clone().detach().requires_grad_().to('cpu')
HL = torch.nn.HuberLoss()
try:
    HL(t1, t2).sum().backward()
except Exception as e:
    print("HuberLoss Fail Backward:", e)

t1 = t1_tensor.clone().detach().requires_grad_().to('cpu')
t2 = t2_tensor.clone().detach().requires_grad_().to('cpu')
SL = torch.nn.SmoothL1Loss()
SL(t1, t2).sum().backward()
print("SmoothL1Loss Succeed Backward")

# HuberLoss Fail Backward: Found dtype Float but expected Double
# SmoothL1Loss Succeed Backward
```

### Versions

pytorch: 1.11.0

### PR DESCRIPTION
**Pull request by qwertyforce**
fixes https://github.com/pytorch/pytorch/issues/76646
summary: backward of huber loss didnt support type promotion and raised an error when different dtypes were used in input

### CODE REVIEW

**Review by soulitzer — APPROVED** [review]
Thanks!

**Comment by qwertyforce** [comment]
Hi @soulitzer! Do you think this is mergeable or is there anything else i can do?
